pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerUtils.java

```diff
@@ -343,4 +349,47 @@ public static boolean isFunctionCodeBuiltin(Function.FunctionDetailsOrBuilder fu
                 .startMessageId(startMessageId)
                 .create();
     }
+
+    public static Producer<byte[]> createExclusiveProducerWithRetry(PulsarClient client,
+                                                                    String topic,
+                                                                    String producerName) {
+        Actions.Action createProducerAction = Actions.Action.builder()
+                .actionName(String.format("Creating exclusive producer for topic %s", topic))
+                .numRetries(5)
+                .sleepBetweenInvocationsMs(10000)
+                .supplier(() -> {
+                    try {
+                        Producer<byte[]> producer = client.newProducer().topic(topic)
+                                .accessMode(ProducerAccessMode.Exclusive)
+                                .enableBatching(false)
```

Comment:
Do you remember if there was any reason for not using batching here?

Reply:
I do not.  We could enable it but I wouldn't expect much benefit as we are not exactly optimizing for metadata operation throughput.  For debugging it is a little easier without batching as you will know exactly how many messages are in the internal topics.